api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -178,11 +261,20 @@ private <T> T submitRequest(String requestPath, String data, Class<T> responseTy
 				out.close();
 			}
 			
-			if (connection.getResponseCode() != 200) {
-				final String error = connection.getResponseCode() + " " + connection.getResponseMessage();
-				throw new APIException("Unexpected response " + error + " from MPI");
-			}
+			responseCode = connection.getResponseCode();
 			
+			if (mpiContext.getMpiSystem().isSanteMPI()) {
+				if (!HTTP_REQUEST_SUCCESS_RANGE.contains(responseCode)) {
+					if (responseCode == 404) {
+						return (T) MapUtils.EMPTY_MAP;
```

Comment:
Why not throw an exception in case of 404?

Reply:
For santeMPI, when you search for non existing patient it return 404. For that case we do not want to throw an exception because there a consecutive step.